submissions/JuliaNovak/frogger-game/app.js

```diff
@@ -0,0 +1,139 @@
+// Enemies our player must avoid
+var Enemy = function (x, y) {
+   this.x = x;
+   this.y = y;
+   // Variables applied to each of our instances go here,
+   // we've provided one for you to get started
+
+   // The image/sprite for our enemies, this uses
+   // a helper we've provided to easily load images
+   this.sprite = "images/enemy-bug.png";
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+   const fieldWidth = 505;
+   // You should multiply any movement by the dt parameter
+   // which will ensure the game runs at the same speed for
+   // all computers.
+   for (let i = 0; i < 200; i++) {
+      this.x += dt;
+      if (this.x > fieldWidth) {
+         this.x = Math.floor(Math.random() * -200);
+      }
+   }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function () {
+   ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now instantiate your Enemy.
+// Place all enemy objects in an array called allEnemies
+const generateEnemies = function (numberOfEnemies) {
+   const allEnemies = [];
+   for (let i = 0; i < numberOfEnemies; i++) {
+      const initialPositionX = 0;
+      const initialPositionY = 63;
+      const yDistanceBetwEnemies = 84;
+      allEnemies.push(
+         new Enemy(
+            initialPositionX,
+            initialPositionY + i * yDistanceBetwEnemies
+         )
+      );
+   }
+   return allEnemies;
+};
+const allEnemies = generateEnemies(3);
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+let Player = function (x, y) {
+   this.x = x;
+   this.y = y;
+   this.sprite = "images/char-cat-girl.png";
+};
+
+const calculatePositions = function (data) {
+   allEnemies.forEach((enemy) => {
+      const xCollisionPoint = 80;
+      const yCollisionPoint = 60;
+      if (
+         data.x < enemy.x + xCollisionPoint &&
+         data.x + xCollisionPoint > enemy.x &&
+         data.y < enemy.y + yCollisionPoint &&
+         yCollisionPoint + data.y > enemy.y
+      ) {
+         data.x = 200;
+         data.y = 400;
+      }
+   });
+};
+Player.prototype.update = function () {
+   calculatePositions(this);
```

Comment:
Make `calculatePositions` a method of an appropiate class. It should follow requirements (re global variables).
There are reasons we have `render`, `handleInput` as methods, not as just global functions we pass class instances to.